hr_simplify/models/hr_simplify.py

```diff
@@ -0,0 +1,107 @@
+##############################################################################
+#
+#    Copyright (C) 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>.
+#    All Rights Reserved.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from odoo import _, api, fields, models
+
+
+class HrEmployee(models.Model):
+    """Simplified Employee Record Interface."""
+
+    _name = "hr.employee"
+    _inherit = "hr.employee"
+    _sql_constraints = [
+        (
+            "unique_identification_id",
+            "unique(identification_id)",
+            _("Official Identifications must be unique!"),
+        ),
+    ]
+
+    @api.model
+    def _default_country(self):
+
+        cid = self.env["res.country"].search([("code", "=", "ET")])
+        if cid:
+            return cid[0]
+
+    country_id = fields.Many2one(default=_default_country)
+    job_id = fields.Many2one(
+        related="contract_id.job_id", comodel_name="hr.job", string="Job", readonly=True
+    )
+
+
+class HrContract(models.Model):
+
+    _inherit = "hr.contract"
+
+    @api.model
+    def _default_employee(self):
+
+        if self.env.context is not None:
+            e_ids = self.env.context.get("search_default_employee_id", False)
+            if e_ids:
+                return e_ids[0]
+
+    employee_id = fields.Many2one(default=_default_employee)
+    employee_dept_id = fields.Many2one(
+        string="Default Dept Id",
+        related="employee_id.department_id",
+        comodel_name="hr.department",
+    )
+
+    @api.onchange("employee_id")
+    def onchange_employee_id(self):
+
+        # dom = { "job_id": [] }
+        if self.employee_id:
+            dept = self.employee_id.department_id
+            self.employee_dept_id = dept.id
+            # dom["job_id"] = [("department_id", "=", dept.id)]
```

Comment:
- What's up with the commented code? Did you add it or was it already there? If it was already there then please just remove it.
- If employee_id is false then employee_dept_id should also be false

Reply:
the commented code was a dynamic domain used to filter Job based on the Department selected and is now deprecated in version 14 to return domain from a onchange method and has no effect on the record filtering. I kept it there as a reminder that should be fixed. a community module called 'web_domain_filed' that implement stored domain could be used